packages/@aws-cdk/aws-cloudfront/lib/web-distribution.ts

```diff
@@ -268,6 +278,13 @@ export interface CustomOriginConfig {
    * @default - No additional headers are passed.
    */
   readonly originHeaders?: { [key: string]: string };
+
+  /**
+   * When you enable Origin Shield in the AWS Region that has the lowest latency to your origin, you can get better network performance
+   *
+   * @default - origin shield not enabled
+   */
+  readonly originShieldRegion?: string;
```

Comment:
Having this property on all three interfaces feels a bit confusing. Is this purely to handle the case where someone wants to create one origin shield region for their primary, and another for their failover?

If so, I'd be slightly more inclined to remove it from the top-level `SourceConfig`; at least that way, there's no ambiguity about whether to use `SourceConfig.originShieldRegion` or `CustomOriginConfig.originShieldRegion`.

Reply:
I do think it would provide a bit more flexibility within origin groups, but you're right it might be at the expense of clarity. In the end I put it on all of them to preserve consistency with `originPath` and `originHeaders`, which are also on all of them. 

All other issues have been addressed in the latest commit, so let me know your thoughts and I can make this adjustment if necessary.

Reply:
Your point about copying the behavior of `originPath` and `originHeaders` is fair. We can keep this as-is.